dev_docs/contributing/documentation.mdx

```diff
@@ -0,0 +1,195 @@
+<DocLink id="kibDevTutorialDebugging" text="Tutorials" /> falls under both _tutorials_ and _how to_. The <DocLink
+  id="kibDevDocsApiWelcome"
+  text="API documentation"
+/> section is _reference_ material.
+
+Developers may choose to keep information that is specific to a particular plugin along side the code.
+
+### Best practices
+
+#### Keep content fresh
+
+A fresh pair of eyes are invaluable. Recruit new hires to read, review and update documentation. Leads should also periodically review documentation to ensure it stays up to date. File issues any time you notice documentation is outdated.
+
+#### Consider your target audience
+
+Documentation in the Kibana Developer Guide is targeted towards developers building Kibana plugins. Keep implementation details about internal plugin code out of these docs.
+
+#### High to low level
+
+When a developer first lands in our docs, think about their journey. Introduce basic concepts before diving into details. The left navigation should be set up so documents on top are higher level than documents near the bottom.
+
+#### Think outside-in
+
+It's easy to forget what it felt like to first write code in Kibana, but do your best to frame these docs "outside-in". Don't use esoteric, internal language unless a definition is documented and linked. The fresh eyes of a new hire can be a great asset.
+
+
+## API documentation
+
+We automatically generate <DocLink id="kibDevDocsApiWelcome" text="API documentation"/>. The following guidelines will help ensure your <DocLink id="kibPlatformIntro" section="public-plugin-api" text="public APIs" /> are useful.
+
+If you encounter an error of the form:
+
+<DocCallOut color="danger" title="JavaScript heap out of memory" />
+
+You can increase [max memory](https://nodejs.org/api/cli.html#--max-old-space-sizesize-in-megabytes) for node as follows:
+
+```bash
+# As a runtime argument
+node --max-old-space-size=8192 foo/bar
+
+# As an env variable, in order to apply it systematically
+export NODE_OPTIONS=--max-old-space-size=8192
+```
+
+### Code comments
+
+Every publicly exposed function, class, interface, type, parameter and property should have a JSDoc style comment.
+
+- Use `@param` tags for every function parameter.
+- Use `@returns` tags for return types.
+- Use `@throws` when appropriate.
+- Use `@beta` or `@deprecated` when appropriate.
+- Use `@removeBy {version}` on `@deprecated` APIs. The version should be the last version the API will work in. For example, `@removeBy 7.15` means the API will be removed in 7.16. This lets us avoid mid-release cycle coordination. The API can be removed as soon as the 7.15 branch is cut.
+- Use `@internal` to indicate this API item is intended for internal use only, which will also remove it from the docs.
```

Comment:
We use `@public` in several places, it's not clear to me what the distinction is, perhaps we should mention that here too?

Reply:
We don't officially recommend using `@public`, it's a convention that I think only the core team uses. I think it'd be great to use, but _only_ if we had ci checks to ensure all public APIs were labeled appropriate, and fail if not. Otherwise it can't be trusted to be accurate.